LeaderboardBackend/Models/Entities/BaseEntity.cs

```diff
@@ -0,0 +1,38 @@
+using System.ComponentModel.DataAnnotations;
+using NodaTime;
+
+/// <summary>
+///     Base class that provides common fields to all Entities.
+/// </summary>
+public class BaseEntity
+{
+	/// <summary>
+	///     The time this entity was created.
+	/// </summary>
+	[Required]
+	public Instant CreatedAt { get; set; } = SystemClock.Instance.GetCurrentInstant();
+
+	/// <summary>
+	///     The time this entity was updated.
+	/// </summary>
+	[Required]
+	public Instant UpdatedAt { get; set; } = SystemClock.Instance.GetCurrentInstant();
+
+	/// <summary>
+	///     The time this entity was deleted.
+	/// </summary>
+	public Instant? DeletedAt { get; set; }
+
+	public override bool Equals(object? obj)
```

Comment:
Make the Equals and GetHashCode also abstract and implemented them in each Entity

Reply:
Actually, I just need to remove these lines in `BaseEntity`, as they already are overrides here. The methods are part of the inner `object` type. I'll still do the latter part, of course.